app/tests/test_training.py

```diff
@@ -648,6 +651,97 @@ def test_difficulty_form(self):
 
                 # should not create any new attempts
                 self.assertEqual(Attempt.query.count(), 1)
+    
+    def test_correct_fill_in_the_blank_question_submission(self):
```

Comment:
I think this test needs to be a bit more rigorous. For example, are we confident that this will work if the answer contains a comma? In the future, someone may get the idea of splitting the string by comma and reporting individual right/wrong parts. We need a test to make sure if they do that, it doesn't mess with answers that have commas in them.